aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -70,8 +70,8 @@
   private static final Cache<Key, SignedComponent> SIGNED_COMPONENT_CACHE =
       Caffeine.newBuilder().expireAfterWrite(30, TimeUnit.SECONDS).maximumSize(100).build();
 
-  private static final ScheduledExecutorService TOKEN_REFRESH_EXECUTOR =
-      ThreadPools.newScheduledPool("s3-signer-token-refresh", 1);
+  private static final AtomicReference<ScheduledExecutorService> TOKEN_REFRESH_EXECUTOR =
```

Comment:
we have been just using `private static volatile` and assign the value once, for example:

https://github.com/apache/iceberg/blob/master/aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java#L81

Is there any advantage for using AtomicReference?

Reply:
the main advantage of using an `AtomicReference` is that you don't need to use `synchronized` & `volatile`, thus the code ends up being shorter and achieves the same thing

Reply:
let me know if you think we should switch to using double-checking locking instead in this class

Reply:
`AtomicReference` is something I've been pushing just because it leads to shorter more readable code Imo but I'm not super opinionated on this, since the rest of the code base follows a good convention we can just follow that